### ISSUE TITLE
Custom sizes for Dalle

### ISSUE THREAD

**Issue by brandonzylstra** [BoltAI/BoltAI#229]
**Is your feature request related to a problem? Please describe.**

If I ask Dalle for a 512x512 image, my request is ignored.  I discovered that this is (at least partly) because sizes are specified in the LLM settings, but the list of options is _extremely limited_.  

<img width="387" alt="image" src="https://github.com/user-attachments/assets/7bda3cd4-e808-4e55-ae24-fd7954e475b2" />

**Describe the solution you'd like**

I would like to be able to specify custom sizes for Dalle images.  (Or any other images that are supported in the future from any other LLM that returns images.)

**Describe alternatives you've considered**

Setting up custom x/y sizes in the LLM settings would be the simplest option.  Ideally these sizes could be global for all image-generating LLMs.  Ideally I could choose the size every time I create an image, instead of having to change preferences every time I want a different size.  And ideally I could also specify one-off sizes when making the request, beyond what options I'd made available to myself in a drop down.

**Comment by longseespace** [comment]
Dall E 3 doesn't support 512x512 images, or custom size unfortunately.

Why do you need 512x512 I'm curious. As for me I think ratio is important, but not the actual size.

![image](https://github.com/user-attachments/assets/59df7272-e861-4a57-8370-afa6a6a7ab9a)

**Comment by brandonzylstra** [comment]
One reason is that Discord recommends 512 x 512 (I'm guessing that's the max size where there's a benefit on a Retina display in whatever context they display the image as the largest ever, but I don't know exactly why they recommend that) but perhaps a bigger reason is that Dalle does a HORRIBLE job following directions, and adds way too much detail even if you tell it not to.  Here's an example:

<img width="68" alt="image" src="https://github.com/user-attachments/assets/db8ca7d9-f900-4c84-85f2-7469d74a11ef" />

The top icon was generated by Dalle.  The next two presumably were by humans.  If Dalle could be forced to use a smaller size than that one problem might be reduced. 

A third reason is that _I'm guessing_ that smaller images will be less resource-intensive to produce and also cost the user less.  When generating only a few images that makes no difference, but given how poorly Dalle follows directions, I could see requesting huge numbers of images every time I need one image.  Over time that cost might add up quite a bit.

But thanks for explaining the limitations—I didn't realize this was a limitation enforced by OpenAI.  In the mean time I'll see if Dalle 2 is any good.  Thanks again!

**Comment by brandonzylstra** [comment]
Actually, there's no way (that I can find) to set the lower sizes with Dall-e 2 in BoltAI.

**Comment by longseespace** [comment]
@brandondrew you're right. But do you really use Dall E 2 😅 I always get feedback that Dall E 2 quality is very low, almost useless for practical use.

**Comment by brandonzylstra** [comment]
yeah, that seems to be true... the only thing appealing to me was the smaller sizes....
